### DISCUSSION TITLE
What is the timeline for upgrading Saltstack beyond 3001.3? Reason I ask is because Qualys is finding vulnerabilities.

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#1996] [Unsupported Versions] [answered]
Hello. Is there a timeline yet for upgrading Saltstack in the current version of Security Onion? I'm asking because Qualys is enumerating  vulnerabilities related to Salt:

> "SaltStack Salt contains the following vulnerabilities:
> CVE-2020-16846: Prevent shell injections in netapi ssh client
> CVE-2020-17490: Prevent creating world readable private keys with the tls execution module.SALT 2019.2.6.
> 
> Affected Versions:
> SaltStack Salt versions prior to SALT 3001.2"

> "SaltStack Salt contains the following vulnerability:
> CVE-2020-25592: eauth was not properly validated when calling Salt SSH via the salt-api.
> 
> Affected Versions:
> SaltStack Salt versions prior to SALT 3001.3"

The current version appears to be 3001.1. 

For me personally, this isn't a priority issue by any measure. I'm running SO and Qualys in a test & learning environment, so we've got no concerns about these findings. But for those running in production, this can be an issue for their audit or compliance groups. Which is why I'm bringing it up.

Is there any rough date set for that upgrade? 

Thanks!

**Comment by dougburks** [comment, marked as answer]
Security Onion 2.3.10 includes the latest SaltStack fixes and we are hoping to release it this week:
https://github.com/Security-Onion-Solutions/securityonion/projects/15

**Reply by ghost** [reply]
Thank you. Much appreciated!

**Reply by dougburks** [reply]
Security Onion 2.3.10 now available!
https://blog.securityonion.net/2020/11/security-onion-2310-now-available.html
